sleap/io/format/nix.py

```diff
@@ -0,0 +1,387 @@
+                if s.name in skel_map:
+                    continue
+                skel_map[s.name] = len(skel_map)
+            return skel_map
+
+        def node_map(source: Labels):
+            n_map = {}
+            for skeleton in source.skeletons:
+                for n in skeleton.nodes:
+                    if n.name in n_map:
+                        continue
+                    n_map[n.name] = len(n_map)
+            return n_map
+
+        def create_feature_array(name, type, block, frame_index_array, shape, dtype):
+            array = block.create_data_array(name, type, dtype=dtype, shape=shape)
+            rd = array.append_range_dimension()
+            rd.link_data_array(frame_index_array, [-1])
+            return array
+
+        def create_positions_array(
+            name, type, block, frame_index_array, node_names, shape, dtype
+        ):
+            array = block.create_data_array(
+                name, type, dtype=dtype, shape=shape, label="pixel"
+            )
+            rd = array.append_range_dimension()
+            rd.link_data_array(frame_index_array, [-1])
+            array.append_set_dimension(["x", "y"])
+            array.append_set_dimension(node_names)
+            return array
+
+        def chunked_write(
+            instances,
+            frameid_array,
+            positions_array,
+            track_array,
+            skeleton_array,
+            pointscore_array,
+            instancescore_array,
+            trackingscore_array,
+            centroid_array,
+            track_map,
+            node_map,
+            skeleton_map,
+            chunksize=10000,
+        ):
+            data_written = 0
+            indices = np.zeros(chunksize, dtype=int)
+            positions = np.zeros((chunksize, 2, len(node_map.keys())), dtype=float)
+            centroids = np.zeros((chunksize, 2), dtype=float)
+            track = np.zeros_like(indices)
+            skeleton = np.zeros_like(indices)
+            pointscore = np.zeros((chunksize, len(node_map.keys())), dtype=float)
+            instscore = np.zeros_like(track, dtype=float)
+            trackscore = np.zeros_like(instscore)
+            dflt_pointscore = [0.0 for n in range(len(node_map.keys()))]
+
+            while data_written < len(instances):
+                print(".", end="")
+                start = data_written
+                end = (
+                    len(instances)
+                    if start + chunksize >= len(instances)
+                    else start + chunksize
+                )
+                for i in range(start, end):
+                    inst = instances[i]
+                    index = i - start
+                    indices[index] = inst.frame_idx
+                    if inst.track is not None:
+                        track[index] = track_map[inst.track.name]
+                    else:
+                        track[index] = -1
+                    skeleton[index] = skeleton_map[inst.skeleton.name]
```

Comment:
Based on the way `track_map` was created (assuming `Track.name` is a unique identifier), it is possible that `inst.track.name` is not in the `track_map`. Likewise for `skeleton_map`.